### PR TITLE
Add maze recursion algorithm and the test file 

### PR DESCRIPTION
**Pull request by hahaong**
### **Describe your change:**

- [x] Add an algorithm?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/main/java/com/thealgorithms/backtracking/MazeRecursion.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+			}
+		}
+	}
+	
+	
+	
+	
+	// Here is another move strategy for the ball: up->right->down->left
+	public static boolean setWay2(int[][] map, int i, int j) {
+		if (map[6][5] == 2) {// means the ball find its path, ending condition
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/MazeRecursion.java:104]: Why [6][5]?

File: src/main/java/com/thealgorithms/backtracking/MazeRecursion.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
+package com.thealgorithms.backtracking;
+
+public class MazeRecursion {
+
+	public static void main(String[] args) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/MazeRecursion.java:5]: Please, remove `main()`

File: src/test/java/com/thealgorithms/backtracking/MazeRecursionTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author onglipwei
+ * @create 2022-08-03 5:17 AM
+ */
+public class MazeRecursionTest {
+
+    @Test
+    public void testforSetWay() {
```
Comment (siriak) [inline src/test/java/com/thealgorithms/backtracking/MazeRecursionTest.java:13]: Please, add a test for `setWay2()`

File: src/main/java/com/thealgorithms/backtracking/MazeRecursion.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	 * @date Jun 23, 202111:36:14 AM
+	 * @param map The maze
+	 * @param i   x coordinate of your ball(target)
+	 * @param j   y coordinate of your ball(target)
+	 * @return If we did find a path for the ball，return true，else false
+	 */
+	public static boolean setWay(int[][] map, int i, int j) {
+		if (map[6][5] == 2) {// means the ball find its path, ending condition
+			return true;
+		} else {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/MazeRecursion.java:75]: `else` clause isn't needed here, just put the code from else block after closing if bracket

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/backtracking/MazeRecursion.java (hunk 1 above)
Comment (hahaong) [reply src/main/java/com/thealgorithms/backtracking/MazeRecursion.java:104]: We set it as a goal, [6][5] is predefined by user as me, it is located at the top right corner of the maze

File: src/test/java/com/thealgorithms/backtracking/MazeRecursionTest.java (hunk 3 above)
Comment (hahaong) [reply src/test/java/com/thealgorithms/backtracking/MazeRecursionTest.java:13]: yes, sure

File: src/main/java/com/thealgorithms/backtracking/MazeRecursion.java (hunk 4 above)
Comment (hahaong) [reply src/main/java/com/thealgorithms/backtracking/MazeRecursion.java:75]: sure

File: src/main/java/com/thealgorithms/backtracking/MazeRecursion.java (hunk 2 above)
Comment (hahaong) [reply src/main/java/com/thealgorithms/backtracking/MazeRecursion.java:5]: sure

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**